UpdateExpiredCgn/table.ts

```diff
@@ -10,24 +12,33 @@ import {
   TableEntry
 } from "../utils/table_storage";
 
+const ExpiredCgnRowKeyType = t.interface({
+  activationDate: Timestamp,
+  expirationDate: Timestamp,
+  fiscalCode: FiscalCode
+});
+
+export type ExpiredCgnRowKeyType = t.TypeOf<typeof ExpiredCgnRowKeyType>;
+
 /**
  * Do something with the user hash extracted from the table entry
  */
-const withFiscalCodeFromEntry = (f: (s: FiscalCode) => void) => (
+const withExpiredCgnRowFromEntry = (f: (s: ExpiredCgnRowKeyType) => void) => (
   e: TableEntry
 ): void => {
   const rowKey = e.RowKey._;
-  return f(rowKey as FiscalCode);
+  // JSON.parse cannot throw cause we are sure rowKey is a valid Json
+  return f(JSON.parse(rowKey) as ExpiredCgnRowKeyType);
 };
 
 /**
  * Fetches all user hashed returned by the provided paged query
  */
 export async function queryUsers(
   pagedQuery: PagedQuery
-): Promise<ReadonlySet<FiscalCode>> {
-  const entries = new Set<FiscalCode>();
-  const addToSet = withFiscalCodeFromEntry(s => entries.add(s));
+): Promise<ReadonlySet<ExpiredCgnRowKeyType>> {
+  const entries = new Set<ExpiredCgnRowKeyType>();
+  const addToSet = withExpiredCgnRowFromEntry(s => entries.add(s));
```

Comment:
If the above is applied:
```suggestion
): Promise<ReadonlySet<ExpiredCgnRowKey>> {
  const entries = new Set<ExpiredCgnRowKey>();
  const addToSet = withExpiredCgnRowFromEntry(s => entries.add(s));
```